src/concepts/pipeline/schemas.md

```diff
@@ -0,0 +1,307 @@
+  avro(Avro schemas)
+  bigquery(BigQuery schemas)
+end
+transpiler[jsonschema-transpiler]
+
+%% hyperlinks
+click json "https://json-schema.org/"
+click avro "https://avro.apache.org/docs/current/spec.html"
+click bigquery "https://cloud.google.com/bigquery/docs/schemas"
+click transpiler "https://github.com/mozilla/jsonschema-transpiler"
+
+%% edges
+json --> transpiler
+transpiler --> avro
+transpiler --> bigquery
+```
+
+### Mozilla Schema Generator
+
+The schema generator will populate schemas with metadata and insert generated sub-schemas at certain paths.
+It uses the probe information service to enumerate map-type fields.
+These fields are converted into a structured column that can be accessed in BigQuery with [Standard SQL](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators).
+Metadata includes fields added during [data ingestion](#data-ingestion) including fields like `submission_timestamp` and `sample_id`.
+
+The main entrypoint is a script that merges and generates `*.schema.json` under the `schemas` directory, then transpiles these to `*.bq`.
+It commits the schema to the `generated-schemas` branch, with a changelog referencing commits in the `master` branch.
+
+## Data Ingestion
+
+### Validation
+
+Data that fails validation is sent to the `payload_bytes_error` table. Each row contains
+an information about the error that caused it, as well as the name of the job associated with it.
+
+```sql
+SELECT
+  document_namespace,
+  document_type,
+  document_version,
+  error_message,
+  error_type,
+  exception_class,
+  job_name
+FROM
+  `moz-fx-data-shared-prod`.payload_bytes_error.telemetry
+WHERE
+  submission_timestamp > TIMESTAMP_SUB(current_timestamp, INTERVAL 1 hour)
+  AND exception_class = 'org.everit.json.schema.ValidationException'
+LIMIT 5
+```
+
+Column | Example Value | Notes
+-|-|-
+`document_namespace` | telemetry |
+`document_type` | main |
+`document_version` | null | The version in the `telemetry` namespace is generated after validation
+`error_message` | `org.everit.json.schema.ValidationException: #/environment/system/os/version: #: no subschema matched out of the total 1 subschemas`
+`error_type` | `ParsePayload` | The `ParsePayload` type is associated with schema validation or corrupt data
+`exception_class` | `org.everit.json.schema.ValidationException `| Java [JSON Schema Validator](https://github.com/everit-org/json-schema) library
+`job_name` | `decoder-0-0121192636-9c56ac6a` | Name of the Dataflow job that can be used to determine the version of the schema artifact
+
+### Decoding
+
+The BigQuery schemas are used to normalize relevant payload data and determine additional
+properties. Normalization involves renaming field names and transforming certain types of data.
+Snake casing is employed across all schemas, which ensures that nested structures can always be
+accessed using the dot operator. Some data must be transformed before insertion, such as map-types
```

Comment:
I don't see how snake casing relates to being able to use the dot operator for field access. Our snake casing implementation does also handle transforming special characters to underscore, but perhaps that should be called out separately.